base/base.cue

```diff
@@ -0,0 +1,159 @@
+package base
+
+deployLabel:    "sourcegraph"
+nameSpaceValue: "ns-sourcegraph"
+
+service: [ID=_]: {
+	apiVersion: "v1"
+	kind:       "Service"
+	metadata: {
+		name: ID
+		labels: {
+			app:    *ID | string
+			deploy: deployLabel
+		}
+		namespace: nameSpaceValue
+		annotations: {
+			"prometheus.io/port":            *"6060" | string
+			"sourcegraph.prometheus/scrape": "true"
+		}
+	}
+	spec: {
+		// Any port has the following properties.
+		ports: [...{
+			port:     int
+			protocol: *"TCP" | "UDP" // from the Kubernetes definition
+			name:     string | *"client"
+		}]
+		selector: metadata.labels // we want those to be the same
+		type:     "ClusterIP"
+	}
+}
+
+deployment: [ID=_]: {
+	apiVersion: "apps/v1"
+	kind:       "Deployment"
+	metadata: {
+		name: ID
+		labels: deploy: deployLabel
+		namespace: nameSpaceValue
+	}
+	spec: {
+		// 1 is the default, but we allow any number
+		replicas: *1 | int
+		template: {
+			metadata: labels: {
+				app: *ID | string
+			}
+			// we always have one namesake container
+			spec: containers: [{name: *ID | string}, ...]
```

Comment:
Is this always true?

Reply:
i relaxed that constraint, i will delete the comment.